src/components/Form.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import apiCall from '../services/api-call';
+import Results from './Results';
+import History from './History';
+import styles from './Form.css';
+
+export default class Form extends Component {
+
+  state = {
+    history: [],
+    apiData: [],
+    url: '',
+    method: '',
+    jsonInput: ''
+  }
+
+  handleChange = ({ target }) => {
+    this.setState({ [target.name]: target.value });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    apiCall(this.state.url, this.state.name)
+      .then(result => {
+        this.setState({ apiData: result });
+        this.setState(state => ({ history: [{ url: this.state.url, method: this.state.method }, ...state.history] }));
```

Comment:
dont have two setStates right after one another. bad things can happen.this.setState(state => ({ 
history: [{ url: this.state.url, method: this.state.method }, ...state.history] ,
apiData: result
}));